crates/common/tedge_config/src/tedge_config_cli/settings.rs

```diff
@@ -374,3 +374,16 @@ impl ConfigSetting for RunPathSetting {
 
     type Value = FilePath;
 }
+
+pub struct FirmwareTimeoutSetting;
+
+impl ConfigSetting for FirmwareTimeoutSetting {
+    const KEY: &'static str = "firmware.timeout";
```

Comment:
@reubenmiller Here I need your insight as it's UX. Which one do you want?

Reply:
I prefer `firmware.timeout` as it is more concise, and `update` does not mean anything to a user, so it will just bring confusion (imho).

Though I can always be persuaded to change my mind if we are using similar style somewhere else (e.g. for consistency).

Reply:
It definitely is concise, but at the same time, I felt that it's a bit too vague. In my head `firmware.timeout` sounds like the timeout for firmware, which doesn't make any sense. But something like `firmware.update.timeout` clearly conveys that it's the timeout for the firmware update operation. When we make the timeout for config management operations also configurable, I'd be adding `config.update.timeout` and `config.snapshot.timeout` which would be consistent with this firmware key.

TBH, I'd have gone for `child.config.update.timeout` to be completely explicit that it is applicable only for child devices, as in future we may want to provide such timeouts for these operations on the tedge device as well (say `tedge.config.update.timeout`). Being explicit now will help us avoid such conflicting/misleading keys in future. But, may be, I'm going too far into the future with this.

Reply:
You make valid points, we definitely need the `child` part the settings, as treating child devices will be different to the main device (when the main device is supported in the future).

My initial thinking for leaving out the `.update` part of the name was because there is only updates possible when firmware. But since other features have multiple actions (`update` and `snapshot`), then I will align with those as well.

So I will take @albinsuresh proposal `firmware.child.update.timeout`.

Example toml

```toml
[firmware]
child.update.timeout = 3600
```